docs/tutorials/custom-id-generator.rst

```diff
@@ -0,0 +1,82 @@
+.. _tutorial-id-generator:
+
+How to define and use a custom ID generator?
+============================================
+
+By default, Kinto uses UUID4 as ``id`` for its records. In some cases, it might
+not be what you want. But you're not out of luck, there is a way to configure
+Kinto to use a different ``id_generator`` than the default one.
+
+Let's define a "name generator", which will output 8 random characters
+(uppercase or lowercase).
+
+.. code-block:: python
+
+    import random
+    import string
+
+    from cliquet.storage import generators
+
+
+    class NameGenerator(generators.Generator):
+
+        regexp = r'^[a-zA-Z0-9-_]{8}$'
+
+        def __call__(self):
+            ascii_letters = ('abcdefghijklmopqrstuvwxyz'
+                             'ABCDEFGHIJKLMOPQRSTUVWXYZ')
+            alphabet = ascii_letters + string.digits + '-_'
+            letters = [random.choice(ascii_letters + string.digits)]
+            letters += [random.choice(alphabet) for x in range(7)]
+            return ''.join(letters)
+
+
+Note that you actually need to define both the generator **and** the validator,
+which can be provided as a regexp.
```

Comment:
There is something important I guess: the id generator should be *random enough* to avoid *id collisions*